library/bignum_mod.h

```diff
@@ -0,0 +1,142 @@
+/**
+ *  Internal bignum functions
+ *
+ *  Copyright The Mbed TLS Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#ifndef MBEDTLS_BIGNUM_MOD_H
+#define MBEDTLS_BIGNUM_MOD_H
+
+#include "common.h"
+
+#if defined(MBEDTLS_BIGNUM_C)
+#include "mbedtls/bignum.h"
+#endif
+
+
+typedef struct
+{
+    size_t n;
+    mbedtls_mpi_uint *p;
```

Comment:
(consistency) we have (size, pointer) here, but (pointer, size) in `mbedtls_mpi_mod_modulus`.

In C it's common to have (pointer, size) (e.g. pointer comes first in memset()), and `struct key_data` in `psa_crypto_core.h` is like this, but other structures, notably `struct mbedtls_mpi`, have (size, pointer).

Given that sizeof(void *) is nearly always >= sizeof(int), but that sizeof(int) can be < sizeof(void *), for best packing of structures it would seem best to start with the pointer.

Either way, it would be better to be consistent

Reply:
This appears to have been addressed in the newly introduced structures `mbedtls_mpi_mod_residue` and `mbedtls_mpi_mod_modulus`, but I agree with that proposal. 

Maybe as part of the optimisation a task should be created to ensure `struct mbedtls_mpi` and other data elements follow a consistent optimal order